vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/Checkbox.java

```diff
@@ -260,4 +261,16 @@ public void setIndeterminate(boolean indeterminate) {
     public boolean isIndeterminate() {
         return isIndeterminateBoolean();
     }
+
+    /**
+     * Replaces the label content with the given label component.
+     *
+     * @param component
+     *            the component to be added to the label.
+     */
+    public void setLabelComponent(Component component) {
```

Comment:
Let's move this method above, after `setLabelAsHtml`. 
Also, please add `@since 23.1` annotation.